lib/galaxy/webapps/galaxy/api/__init__.py

```diff
@@ -104,3 +106,42 @@ class BaseGalaxyAPIController(BaseAPIController):
 
     def __init__(self, app: StructuredApp):
         super().__init__(app)
+
+
+class Router(InferringRouter):
+    """A FastAPI Inferring Router tailored to Galaxy.
+    """
+
+    def get(self, *args, **kwd):
```

Comment:
The base router object also defines methods to handle patch, trace, head and options HTTP request methods. Although we might not be using all of those right now, we might at some point, and having them accept `require_admin` (or any other galaxy-specific keyword args) will keep them all consistent with the overridden get/post/put/delete. I think it'll be convenient and might save us from some future debugging. What do you think?

Reply:
I tend toward not adding code we don't use and wouldn't have test coverage - and in fact I didn't have delete and put in here and Galaxy just failed to start with a very clear error message so it was pretty clear what the issue was so I wouldn't worry too much about debugging the problem. But I can definitely see implementing a complete solid, consistent interface as being important and I'll happily add those if you'd like.

Reply:
Now that you mention this, even though both have merit (not adding code we don't use, and having a complete interface), I think you're right - let's not add them for now: it's not a framework after all. Thanks for considering this!

Reply:
It is a very good point, thanks for bringing it up!